src/internal/pachd/builder.go

```diff
@@ -451,8 +451,8 @@ func (b *builder) startPFSMaster(ctx context.Context) error {
 func (b *builder) startPPSWorker(ctx context.Context) error {
 	etcdPrefix := path.Join(b.env.Config().EtcdPrefix, b.env.Config().PPSEtcdPrefix)
 	w := pps_server.NewWorker(pps_server.WorkerEnv{
-		TaskService:   b.env.GetTaskService(etcdPrefix),
-		GetPachClient: b.env.GetPachClient,
+		TaskService: b.env.GetTaskService(etcdPrefix),
+		PFS:         b.env.GetPachClient(ctx).PfsAPIClient,
```

Comment:
That's just the only way to get a pach client, which I only want to pull out the PFS client.  It has an internal context which isn't a great model, and we are trying to move away from it.  In my no-service-env branch, this is not required to setup the workers.

I don't want to pass around a `*grpc.ClientConn` because it doesn't communicate any information about what is supposed to be listening on the other side.  In the case of the sidecar and pachw, only a subset of all the services are listening on local gRPC.  If someone needs more APIs in the PPS worker, I want it to be obvious that it's a bigger lift than just calling `otherapi.NewAPIClient`.  They will also have to setup the server end of that in the worker, which might cause them to reconsider pulling in all those dependencies.

Reply:
That's reasonable.  I do think that implementing a fake PFS client is what we want for testing, anyway.  Having a clientconn that you don't is weird in that case.